Entities/AHuman.cpp

```diff
@@ -1796,6 +1803,39 @@ bool AHuman::UpdateMovePath()
     return true;
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+
+void AHuman::UpdateWalkAngle(AHuman::Layer whichLayer) {
+	if (m_Controller.IsState(BODY_JUMP)) {
+		m_WalkAngle[whichLayer] = Matrix(c_QuarterPI * GetFlipFactor());
+	} else {
+		// Cast rays to calculate the approximate shape of terrain.
+		int rayCount = 4;
+		float rayLength = 10.0F;
```

Comment:
But it does use the leg length, `10.0F` is just the base value since the actor will ideally want to see a bit further than its leg so that the angle can update correctly before the next step

Reply:
Ohh yeah, just me being blind again I guess. Cool.